aomi/filez.py

```diff
@@ -0,0 +1,237 @@
+
+    for user in config.get('users', []):
+        pfile = user['password_file'],
+        freeze_secret(pfile, pfile, 'user', tmp_dir, opt)
+
+    for secret in config.get('secrets', []):
+        if 'var_file' in secret:
+            sfile = secret['var_file']
+            freeze_secret(sfile, sfile, 'var_file', tmp_dir, opt)
+        elif 'aws_file' in secret:
+            sfile = secret['aws_file']
+            freeze_secret(sfile, sfile, 'aws_file', tmp_dir, opt)
+        elif 'files' in secret:
+            for a_secret in secret['files']:
+                sfile = a_secret['source']
+                freeze_secret(sfile, sfile, 'file', tmp_dir, opt)
+
+
+def freeze_archive(tmp_dir, dest_prefix):
+    """Generates a ZIP file of secrets"""
+    zip_filename = "%s/aomi-blah.zip" % tmp_dir
+    archive = zipfile.ZipFile(zip_filename, 'w')
+    for archive_file in os.listdir(dest_prefix):
+        archive.write("%s/%s" % (dest_prefix, archive_file), archive_file)
+
+    archive.close()
+    return zip_filename
+
+
+def freeze_encrypt(dest_dir, zip_filename, tmp_dir, config, opt):
+    """Encrypts the zip file"""
+    gpg = gnupg.GPG(homedir="%s/.gnupg" % os.environ['HOME'],
+                    binary='/usr/local/bin/gpg')
+
+    pgp_keys = grok_keys(gpg, config, opt)
+
+    handle = open(zip_filename, 'r')
+    zip_data = handle.read()
+    handle.close()
+    shutil.rmtree(tmp_dir)
+    ice_handle = os.path.basename(os.path.dirname(opt.secretfile))
+    timestamp = time.strftime("%H%M%S-%m-%d-%Y",
+                              datetime.datetime.now().timetuple())
+    ice_file = "%s/aomi-%s-%s.ice" % (dest_dir, ice_handle, timestamp)
+    enc_resp = gpg.encrypt(zip_data,
+                           *pgp_keys,
+                           output=ice_file)
+    if not enc_resp.ok:
+        problems("Unable to encrypt zipfile")
+
+    return ice_file
+
+
+def freeze(dest_dir, opt):
+    """Iterates over the Secretfile looking for secrets to freeze"""
+    if not (os.path.exists(dest_dir) and
+            os.path.isdir(dest_dir)):
+        os.mkdir(dest_dir)
+
+    tmp_dir = tempfile.mkdtemp('aomi-freeze')
+    dest_prefix = "%s/dest" % tmp_dir
+    os.mkdir(dest_prefix)
+    config = get_secretfile(opt)
+
+    freeze_files(config, tmp_dir, opt)
+    zip_filename = freeze_archive(tmp_dir, dest_prefix)
+    ice_file = freeze_encrypt(dest_dir, zip_filename, tmp_dir, config, opt)
+    print("Generated file is %s" % ice_file)
+
+
+def thaw_decrypt(src_file, tmp_dir, opt):
+    """Decrypts the encrypted ice file"""
+    gpg = gnupg.GPG(homedir="%s/.gnupg" % os.environ['HOME'],
+                    binary='/usr/local/bin/gpg')
+
+    if not os.path.isdir(opt.secrets):
+        warning("Creating secret directory %s" % opt.secrets)
+        os.mkdir(opt.secrets)
+
+    zip_file = "%s/aomi.zip" % tmp_dir,
+    ice_handle = open(src_file, 'r')
+    resp = gpg.decrypt_file(ice_handle, output=zip_file)
```

Comment:
Maybe pass this 144 and 153 in as a function like

```
def simple_decrypter(homedir, binary):
    gpg= gnupg.GPG(homedir=homedir, binary = binary)
    lambda(in, out) :
        resp = gpg.decrypt_file(in, output=out)


      
def thaw_decrypt(src_file, tmp_dir, decrypter, opt):
      ....

```
This would probably make testing much simpler.